client/src/components/NotificationPopUp/NotificationPopUp.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import useStyles from './useStyles';
+import NotificationsNoneIcon from '@material-ui/icons/NotificationsNone';
+import Badge from '@material-ui/core/Badge';
+import Popover from '@material-ui/core/Popover';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import Avatar from '@material-ui/core/Avatar';
+import { withStyles } from '@material-ui/core/styles';
+import { getNotification, updateNotification } from '../../helpers/APICalls/notification';
+import { Notification } from '../../interface/User';
+
+const GlobalCss = withStyles({
+  // @global is handled by jss-plugin-global.
+  '@global': {
+    // You should target [class*="MuiButton-root"] instead if you nest themes.
+    '.MuiPopover-paper': {
+      paddingBottom: 40,
+    },
+    '*::-webkit-scrollbar': {
+      width: '0px',
+    },
+  },
+})(() => null);
+
+const NotificationPopUp = (): JSX.Element => {
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  useEffect(() => {
+    async function getAll() {
+      const allNotifications = await getNotification();
+      if (allNotifications) {
+        setNotifications(allNotifications);
+      } else {
+        console.log('Notifications not found');
+        new Error('Notifications not found');
+      }
+    }
+    getAll();
+  }, [id]);
+
+  const filterNotification = notifications.length
+    ? notifications?.filter((notification) => notification.opened === false)
+    : [];
+
+  const handleNotification = (event: React.MouseEvent<HTMLButtonElement>) => {
+    // setOpen(true);
+    /**
+     *  changing the opened property to true by sending post request
+     *  so that once the notifications navitem is clicked, those notifications would not be counted as new
+     */
+    filterNotification.length &&
+      filterNotification?.map(async (notification) => await updateNotification(notification));
+    setAnchorEl(event.currentTarget);
+  };
+
+  const hoursCalculator = (createdAt: string): string => {
```

Comment:
Good thinking with this 👍🏽 

But we have moment.js installed, don't we? You can use the functions within moment.js to make your life easy.

Reply:
I realized it later :), should i leave like this for now?

Reply:
Yes, you can it leave it for now :)